lib/tandem/tandemSimulator.js

```diff
@@ -152,13 +152,36 @@ exports.make = function(config){
         setCurrTempBasal(event);
       }
       else if(event.subType === 'stop') {
-        //add time_left in case temp basal was terminated early and it's the final basal
         if(currTempBasal != null) {
           var tempBasal = currTempBasal;
           tempBasal.subType = 'stop';
           tempBasal.time_left = event.time_left;
           setCurrTempBasal(tempBasal);
         }
+
+        if (currBasal !== null) {
+          if (currBasal.deliveryType === 'temp' && currTempBasal && currTempBasal.time_left) {
+          // check if temp basal was cancelled
+            if (currTempBasal.time_left !== 0 ) {
+              var duration;
+              if (currBasal.isAssigned('duration')) {
+                // a new-day event during a cancelled temp basal
```

Comment:
I don't quite follow the leap of logic in this comment. Why is this necessarily a _new-day_ event?